### PR TITLE
Chore: re-arrange folders and files

### PR DESCRIPTION
**Pull request by yaronya**
### Feature Request
Felt like the current project structure is a mess.
Additionally, splitting that gigantic `env0-deploy-flow` file into separate isolated files/functions is the way-to-go 💯  

### Solution
Split the project into folders:
1. `commands` - commands logic is kept here.
2. `utils` - utilities that are used by commands are kept here.
3. `config` - some global configurations of the CLI are kept here (like commands configuration, arguments and etc.)
4. `main.js` the entrypoint of the CLI.

### CODE REVIEW

File: node/src/utils/assertions.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { options } = require('../config/constants');
+
+const { API_KEY, API_SECRET, ORGANIZATION_ID, PROJECT_ID, ENVIRONMENT_NAME } = options;
+
+const assertDeploymentStatus = status => {
```
Comment (avnerenv0) [inline node/src/utils/assertions.js:5]: This is business logic, so I wouldn't put it in a general `util`.
IMO `assertDeploymentStatus` is part of `deploy-utils`,
`assertRequiredOptions` is part of whatever is managing the configuration.

File: node/src/main.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    if (error.response && error.response.data && error.response.data.message) {
+      message += `: ${error.response.data.message}`;
+    }
+
+    console.error(message);
+    process.exit(1);
+  }
+};
+
+const parseEnvironmentVariables = (environmentVariables, sensitive) => {
```
Comment (avnerenv0) [inline node/src/main.js:68]: IMO `parseEnvironmentVariables` and `getEnvironmentVariablesOptions` should be part of whatever is managing the configuration

File: node/src/main.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    assertCommandExists(command);
+
+    const commandDefinitions = commands[command].options;
+    const commandOptions = commandLineArgs(commandDefinitions, { argv });
+    const environmentVariables = getEnvironmentVariablesOptions(
+      commandOptions[ENVIRONMENT_VARIABLES],
+      commandOptions[SENSITIVE_ENVIRONMENT_VARIABLES]
+    );
+
+    await runCommand(command, commandOptions, environmentVariables);
```
Comment (avnerenv0) [inline node/src/main.js:55]: IMO `environmentVariables` should be a part of `commandOptions`

File: node/src/utils/assertions.js (hunk 1 above)
Comment (yaronya) [reply node/src/utils/assertions.js]: 3608b42

File: node/src/main.js (hunk 3 above)
Comment (yaronya) [reply node/src/main.js:55]: I agree, raw environmentVariables attribute should be overridden.
I'll make this change in a following PR.

File: node/src/main.js (hunk 2 above)
Comment (yaronya) [reply node/src/main.js:68]: I agree, they shouldn't exist on `main.js` but couldn't find where to put them now.
I'll make this change in a following PR.

**Comment by yaronya** [comment]
@avnerenv0 I didn't like that `utils` folder name either. I've changed it to `lib` for now but be sure that this structure isn't perfect nor intended to be 🥈 
I hope we'll find some time to organize even more a lil bit.

**Pull request merged**